classes/rest/class-rest-api.php

```diff
@@ -17,46 +18,97 @@ class Rest_Api {
 	 * Add custom endpoints.
 	 */
 	public static function add_endpoints(): void {
-		add_action(
-			'rest_api_init',
-			static function () {
-				/**
-				 * A lightweight endpoint to get all posts with only id and title.
-				 */
-				register_rest_route(
-					self::REST_NAMESPACE,
-					'/all-published-posts',
-					[
-						[
-							'methods'  => WP_REST_Server::READABLE,
-							'callback' => static function () {
-								global $wpdb;
-
-								if ( is_plugin_active( 'sitepress-multilingual-cms/sitepress.php' ) ) {
-									// This is public data, no nonce needed.
-									// phpcs:ignore WordPress.Security.NonceVerification.Recommended
-									if ( ! isset( $_GET['post_language'] ) ) {
-										return new \WP_REST_Response(
-											'WPML is active so you need to query posts with the `post_language` query parameter.',
-											400
-										);
-									}
-									// This is public data, no nonce needed.
-									// phpcs:ignore WordPress.Security.NonceVerification.Recommended
-									$query = self::get_wpml_posts_query( $_GET['post_language'] );
-								} else {
-									$query = "SELECT id, post_title FROM wp_posts WHERE post_status = 'publish' AND post_type = 'post' ORDER BY post_date DESC";
-								}
-								// The query is prepared, just not in this line.
-								// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
-								$result = $wpdb->get_results( $query );
-
-								return rest_ensure_response( $result );
-							},
-						],
-					]
-				);
-			}
+		add_action( 'rest_api_init', [ __CLASS__, 'endpoints']);
+
+	}
+
+	public static function endpoints() {
+		/**
+		 * A lightweight endpoint to get all posts with only id and title.
```

Comment:
Is this comment correct? Looks vestigial.

Reply:
It's lightweight in the sense that getting the same 2 fields for all posts from the wordpress API would mean more overhead. This is guaranteed to be fast even when there is a large amount of posts.